compiler-rt/lib/tysan/tysan_interceptors.cpp

```diff
@@ -22,6 +22,12 @@
 #define TYSAN_INTERCEPT___STRDUP 0
 #endif
 
+#if SANITIZER_POSIX
+#define TYSAN_INTERCEPT___STRNDUP 1
+#else
+#define TYSAN_INTERCEPT___STRNDUP 0
+#endif
+
```

Comment:
Could the conditions for intercepting STRNDUP be merged with STRDUP or do we need different conditions?

Reply:
yes strndup is not supported on windows (which tysan seems to support minus the tests), while strdup everything should support it.